### ISSUE TITLE
Multiple sessions with per_process_gpu_memory_fraction

### ISSUE THREAD

**Issue by sealedtx** [tensorflow/tensorflow#32919]
**System information**
- OS Platform and Distribution: Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): 1.14.0
- TensorFlow version (use command below): v1.14.0-rc1-22-gaf24dc91b5 1.14.0
- Python version: 3.6.8
- CUDA/cuDNN version: 10.0 / 7.4.2.24-1
- GPU model and memory: GTX 1070ti, 8119Mb

**Describe the current behavior**
Start multiple sessions with gpu_options and different per_process_gpu_memory_fraction, but tensorflow outputs same amount of memory for both sessions:
```
...
tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcuda.so.1
tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5a67330 executing computations on platform CUDA. Devices:
tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): GeForce GTX 1070 Ti, Compute Capability 6.1
tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3696000000 Hz
tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5a645c0 executing computations on platform Host. Devices:
tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>
tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: 
name: GeForce GTX 1070 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.683
pciBusID: 0000:01:00.0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7
tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0
tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:
tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 
tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N 
tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 500 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1070 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)

... init second ...

tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0
tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7
tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0
tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:
tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 
tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N 
tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 500 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1070 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
```

**Describe the expected behavior**
I expect each session to allocate amount of memory specified in config.gpu_options

```
...
Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 500 MB memory)
...
Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 300 MB memory)
```

**Code to reproduce the issue**
    
    def init_first(self):
        graph_def = tf.compat.v1.GraphDef()
        with tf.io.gfile.GFile(self.MODEL_PATH[0], 'rb') as f:
            graph_def.ParseFromString(f.read())

        graph = tf.Graph()
        with graph.as_default():
            tf.import_graph_def(graph_def, name='import')

        input = graph.get_tensor_by_name('import/image_tensor:0')
        output = [
            graph.get_tensor_by_name('import/boxes:0'),
            graph.get_tensor_by_name('import/scores:0'),
        ]

        config = tf.compat.v1.ConfigProto()
        config.gpu_options.allow_growth = True
        config.gpu_options.per_process_gpu_memory_fraction = 0.061
        sess = tf.compat.v1.Session(graph=graph, config=config)
        return sess, input, output

    def init_second(self):
        graph_def = tf.compat.v1.GraphDef()
        with tf.io.gfile.GFile(self.MODEL_PATH[1], 'rb') as f:
            graph_def.ParseFromString(f.read())

        graph = tf.Graph()
        with graph.as_default():
            tf.import_graph_def(graph_def)

        config = tf.compat.v1.ConfigProto()
        config.gpu_options.allow_growth = True
        config.gpu_options.per_process_gpu_memory_fraction = 0.37
        sess = tf.compat.v1.Session(graph=graph, config=config)

        output = ('import/softmax_2/softmax:0',
                  'import/conv6-2/conv6-2:0',
                  'import/conv6-3/conv6-3:0')
        input = 'import/Placeholder_2:0'

        return sess, input, output

    firsrt = init_first()
    second = init_second()

**Comment by gadagashwini** [comment]
@sealedtx, Please try with latest Tf 1.15.0.rc1 version. And also provide the complete code to reproduce the issue . Looks like MODEL_PATH is not given. Thanks!

**Comment by sealedtx** [comment]
@gadagashwini I have tested latest Tf 1.15.0.rc1 version and it has same behavior. MODEL_PATH is just path to my custom networks files .pb

**Comment by ppwwyyxx** [comment]
This is an ancient bug #8136 which is unlikely to be fixed. Once the GPU devices are created, its configuration cannot be changed any more.

**Comment by sealedtx** [comment]
@ppwwyyxx So it is not possible to manage gpu_memory_fraction for multiple sessions in a single process?

**Comment by ppwwyyxx** [comment]
I think that's true.

**Comment by sealedtx** [comment]
Sad to hear this :( 
Really need this feature. Should I close the issue or leave as "want enchantment"?

**Comment by aaroey** [comment]
I'm closing this, please see #8136 for more info.
